### PR TITLE
Add GCF Python SQL samples

### PR DESCRIPTION
**Pull request by ace-n**
~~**Do not merge** until CI is set up.~~

@stew-r FYI

~~@andrewsg FYI - `psycopg2` gives [a warning about binary modules](http://initd.org/psycopg/docs/install.html#binary-install-from-pypi). I assume that we can ignore that?~~

### CODE REVIEW

**Comment by ace-n** [comment]
TODO Add tests/CI

**Comment by andrewsg** [comment]
Any idea why Travis build is failing?

File: functions/sql/mysql_sample.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+# Create SQL connection globally to enable reuse
+mysql_connection = pymysql.connect(**mysql_config)
+
+
+def mysql_demo(request):
+    with mysql_connection.cursor() as cursor:
+        cursor.execute('SELECT NOW() as now')
+        results = cursor.fetchone()
+        return str(results['now'])
```
[the remark this answers is not in the document]
Comment (ace-n) [reply functions/sql/mysql_sample.py]: These are HTTP-triggered functions - `print` statements show up in Cloud Logging, while `return` returns the contents as a response.

[the remark this answers is not in the document]
Comment (andrewsg) [reply functions/sql/mysql_sample.py]: Whoops, I forgot about that context. You're right, we should keep the return.

File: functions/sql/mysql_test.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import mock
+
+env_vars = {
+  'SQL_USER': os.getenv('MYSQL_USER'),
+  'SQL_PASSWORD': os.getenv('MYSQL_PASSWORD'),
+  'SQL_NAME': os.getenv('MYSQL_NAME'),
+  'INSTANCE_CONNECTION_NAME': os.getenv('MYSQL_INSTANCE')
+}
+
+date_regex = re.compile('\d{4}-\d{1,2}-\d{1,2}\s\d{1,2}:\d{1,2}:\d{1,2}')
```
[the remark this answers is not in the document]
Comment (ace-n) [reply functions/sql/mysql_test.py]: I'm not familiar with the details of Python SQL client libraries. If the query fails, are we confident they'll send back an exception?

If so, I don't mind changing this.

(Alternative compromise: I'd _prefer_ to use a shorter regex, or some date-parsing library [`dateutil` maybe?]. That way, we reduce the likelihood of "silent failures" that are common in Node.js.)

File: testing/test-env.tmpl.sh [hunk 3]
```diff
@@ -1,12 +1,24 @@
...
+export GOOGLE_CLOUD_PROJECT=$GCLOUD_PROJECT
+export FIRESTORE_PROJECT=
 export CLOUD_STORAGE_BUCKET=$GCLOUD_PROJECT
 export API_KEY=
 export BIGTABLE_CLUSTER=bigtable-test
 export BIGTABLE_ZONE=us-central1-c
 export SPANNER_INSTANCE=
 export COMPOSER_LOCATION=us-central1
 export COMPOSER_ENVIRONMENT=
+export CLOUD_KMS_KEY=
```
[the remark this answers is not in the document]
Comment (ace-n) [reply testing/test-env.tmpl.sh]: Added @SurferJeffAtGoogle as a reviewer 👍

File: functions/sql/mysql_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewsg) [reply functions/sql/mysql_test.py]: It will definitely send back an exception if it fails.

**Comment by ace-n** [comment]
RE Travis build failure: Travis doesn't have the Cloud SQL proxy enabled. I believe we're intending to deprecate Travis in favor of Kokoro, but I don't know what the timeline on that is.

@SurferJeffAtGoogle should I disable this test (on Travis) for now and move it to Kokoro, or will y'all take care of that later on?

**Comment by ace-n** [comment]
TODO add [best practices](https://cloud.google.com/functions/docs/sql#best_practices_for_cloud_sql_in_cloud_functions) samples.

File: functions/sql/mysql_sample.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START functions_sql_mysql]
+from os import getenv
+
+import pymysql
+
+is_production = getenv('SUPERVISOR_HOSTNAME') is not None
```
[the remark this answers is not in the document]
Comment (ace-n) [reply functions/sql/mysql_sample.py]: The Node samples use the `NODE_ENV` env var to determine whether a function has been deployed. For the sake of maintaining patterns across samples, is there a way we can do this here?

File: testing/test-env.tmpl.sh [hunk 5]
```diff
@@ -1,13 +1,25 @@
...
+
+export MYSQL_INSTANCE=
+export MYSQL_USER=
+export MYSQL_PASSWORD=
+export MYSQL_DATABASE=
+export POSTGRES_INSTANCE=
+export POSTGRES_USER=
+export POSTGRES_PASSWORD=
+export POSTGRES_DATABASE=
```
Comment (SurferJeffAtGoogle) [inline testing/test-env.tmpl.sh]: We haven't standardized these names across languages/repos, and these names look fine.

**Pull request merged**